docs/reference-guide/storage-modes.md

```diff
@@ -117,4 +138,119 @@ In the `IN_MEMORY_ANALYTICAL` mode, periodic snapshots are **disabled**.
+Therefore, the in-memory storage mode uses a pessimistic approach and
+immediately checks whether there is a conflict between two transactions. 
+
+In the on-disk storage mode, the cache is used per transaction. This
+significantly simplifies object management since there is no need to question
+certain object's validity, but it also requires the optimistic approach for
+conflict resolution between transactions. 
+
+In the on-disk storage mode, the conflict is checked at the transaction's commit
+time with the help of RocksDB's transaction support. This also implies that
+deltas are cleared after each transaction, which can optimize memory usage
+during execution. Deltas are still used to fully support Cypher's semantic of
+the write queries. The design of the on-disk storage also simplifies the process
+of garbage collection, since all the data is on disk.
+
+### Isolation level
+
+The on-disk storage mode supports only snapshot isolation level. Mostly because
+it's the Memgraph viewpoint that snapshot isolation should be the default
+isolation level for most applications relying on databases. But the snapshot
+isolation level also simplifies the query's execution flow since there is no
+need to go on disk until the commit of the transaction.
+
+### Indices
+
+Disk-storage supports both label and label-property indices. They are stored in separate
+RocksDB instances as key-value pairs so that the access to the data is faster. Whenever
+the indexed vertex is accessed, it is put into the separate in-memory cache so that
+the reading speed is maximized.
+
+### Constraints
+
+The on-disk storage mode supports both existence and uniqueness constraints.
+Existence constraints don't use context from the disk since the validity of
+nodes can be checked by looking only at this single node. On the other side,
+uniqueness constraints require a different approach. For a node to be valid, the
+engine needs to iterate through all other nodes under constraint and check
+whether a conflict exists. To speed up this iteration process, nodes under
+constraint are stored into a separate RocksDB instance to eliminate the cost of
+iterating over nodes which are not under constraint.
+
+### Data formats
+
+Vertex format for main disk storage:
+Key - `label1, label2, ... | vertex gid | commit_timestamp`
+Value - `property1, property2`
+
+Edge format for main disk storage:
+Key - `from vertex gid | to vertex gid | 0 | edge type | edge gid | commit_timestamp`
+Value - `property1, property2`
+`0` is a placeholder for edge direction in future.
+
+Format for label index on disk:
+
+Key - `indexing label | vertex gid | commit_timestamp`
+
+Value - `label1_id, label2_id, ... | property1, property2, ...`
+
+Value does not contain `indexing label`.
+
+Format for label-property index on disk:
+
+Key - `indexing label | indexing property | vertex gid | commit_timestamp`
+
+Value - `label1_id, label2_id, ... | property1, property2, ...`
+
+Value does not contain `indexing label`.
```

Comment:
This requires some intro and explanation... what am I looking at?

Reply:
Format how data is serialized to the disk. Data = vertices and edges.